models/classes/class.QtiTestService.php

```diff
@@ -570,6 +570,22 @@ protected function importTest(core_kernel_classes_Class $targetClass, Resource $
                 // If any, assessmentSectionRefs will be resolved and included as part of the main test definition.
                 $testDefinition->includeAssessmentSectionRefs(true);
 
+                if($overwriteTest) {
+                    foreach ($testClass->getInstances() as $testInstance) {
+                        if($testInstance->getLabel() === $testDefinition->getDocumentComponent()->getTitle()) {
+                            $this->getTestService()->deleteTest($testInstance);
+                        }
+                    }
+
+                    foreach ($itemClass->getSubClasses() as $subClass) {
+                        if($subClass->getLabel() === $testDefinition->getDocumentComponent()->getTitle()) {
+                            $this->getItemTreeService()->deleteClass($subClass);
+                        }
+                    }
+                }
```

Comment:
We can optimize performance by avoiding call service initialization multiple times. Even for the singleton cases, we still do some unnecessary repeated process.

```suggestion
                $testService = $this->getTestService();
                $itemTreeService = $this->getItemTreeService();
                
                if ($overwriteTest) {
                    foreach ($testClass->getInstances() as $testInstance) {
                        if ($testInstance->getLabel() === $testDefinition->getDocumentComponent()->getTitle()) {
                            $testService->deleteTest($testInstance);
                        }
                    }

                    foreach ($itemClass->getSubClasses() as $subClass) {
                        if ($subClass->getLabel() === $testDefinition->getDocumentComponent()->getTitle()) {
                            $itemTreeService->deleteClass($subClass);
                        }
                    }
                }
```

Reply:
fixed